php/libraries/Subproject.class.inc

```diff
@@ -0,0 +1,109 @@
+<?php declare(strict_types=1);
+/**
+ * This file contains a class which encapsulates the concept of a "Subproject"
+ * in LORIS.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+/**
+ * The Project class encapsulates all details of a project.
+ *
+ * @category Main
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Subproject implements \JsonSerializable
+{
+    private $_subproject_id;
+    private $_project_id;
+    private $_title;
+    private $_windowDifference;
+    private $_recruitmentTarget;
+
+    /**
+     * Contructor
+     *
+     * @param string     $id                The subproject id
+     * @param \ProjectID $project_id        the associated project id
+     * @param string     $title             This title
+     * @param ?string    $windowDifference  Value from the database
+     * @param ?string    $recruitmentTarget A numeric string or null
+     */
+    public function __construct(
+        string $id,
+        \ProjectID $project_id,
+        string $title,
+        ?string $windowDifference = null,
+        ?string $recruitmentTarget = null
+    ) {
+        $this->_subproject_id    = $id;
+        $this->_project_id       = $project_id;
+        $this->_title            = $title;
+        $this->_windowDifference = $windowDifference;
+        if (!is_null($recruitmentTarget)) {
+            $this->_recruitmentTarget = intval($recruitmentTarget);
+        }
+    }
+
+    /**
+     * Gets the visit names for that subproject
+     *
+     * @return array An array of Visit object
+     */
+    public function getVisits(): array
+    {
+        $factory    = \NDB_Factory::singleton();
+        $visitsData = $factory->database()->pselect(
```

Comment:
The constructor should probably have a `\LORIS\LorisInstance` in order to avoid singletons

Reply:
@driusan In addition to the properties of a subproject? Should there be 2 classes then? One to "manage" subprojects and one that contains a subproject properties/data.

Reply:
if the class depends on it it should be injected somewhere. An alternative would be to make it an argument to `getVisits()` since it's getVisits that depends on it.

Reply:
I dont like passing non-DTO stuff in the constructor, mixes concerns

Reply:
I'm going with the controller solution

controller/DAO + DTO